pkg/aks/client.go

```diff
@@ -133,5 +142,39 @@ func GetSecrets(secretsCache wranglerv1.SecretCache, spec *aksv1.AKSClusterConfi
 	cred.AuthBaseURL = spec.AuthBaseURL
 	cred.BaseURL = spec.BaseURL
 
+	if cred.TenantID == "" {
+		cred.TenantID, err = GetCachedTenantID(cred.SubscriptionID, secret)
+		if err != nil {
+			return nil, err
+		}
+		secret.Annotations[tenantIDAnnotation] = cred.TenantID
+		secret.Annotations[tenantIDTimestampAnnotation] = time.Now().UTC().Format(time.RFC3339)
+		_, err = secretClient.Update(secret)
```

Comment:
Should only update these if timestamp was expired and value was not from annotation.